dependency/engine.go

```diff
@@ -425,14 +429,20 @@ func (engine *Engine) requestStart(name string, delay time.Duration) {
 		delay = time.Duration(float64(delay) * fuzz).Round(time.Millisecond)
 	}
 
-	go engine.runWorker(name, delay, manifold.Start, context)
+	go engine.runWorker(name, delay, manifold.Start, snapshot)
 }
 
-// context returns a context backed by a snapshot of current
-// worker state, restricted to those workers declared in inputs. It must only
-// be called from the loop goroutine; see inside for a detailed discussion of
-// why we took this approach.
-func (engine *Engine) context(name string, inputs []string, abort <-chan struct{}) *context {
+// scopedContext returns a context that will be tied to the lifecycle of
+// the tomb.
+func (engine *Engine) scopedContext() (context.Context, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return engine.tomb.Context(ctx), cancel
```

Comment:
reading the engine.tomb.Context() documentation, wouldn't it make more sense to just use engine.tomb.Context() ?
I guess you're trying to get access to its cancel function.

Reply:
Correct.